### PR TITLE
Karmel backup

### PR DESCRIPTION
**Pull request by karmelyoei**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,11 @@ package-lock.json
...
 
 .env
+
+.vscode
+
+<<<<<<< HEAD
+.vscode/launch.json
+=======
+.vscode/launch.json
+>>>>>>> karmel-backup
```
Comment (leggomuhgreggo) [inline .gitignore]: Looks like a small merge conflict here.

```suggestion
.vscode/launch.json
```

File: client/.gitignore [hunk 2]
```diff
@@ -1,4 +1,6 @@
 # See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+.vscode/launch.json
+
```
Comment (leggomuhgreggo) [inline client/.gitignore]: Does the client have a separate git history? If not this should be in the root .gitignore

File: client/src/Components/About/index.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import React, { Component, Fragment } from 'react';
+import style from './style';
+import { Button, Card, Grid, Typography } from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import { Link as RouterLink } from 'react-router-dom';
+class About extends Component {
+	state = {};
+	handleClick = (event) => {
+		event.preventDefault();
+	};
```
Comment (leggomuhgreggo) [inline client/src/Components/About/index.js]: Small point but I think we can get rid of these.

File: client/src/Components/About/index.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+									</Typography>
+
+									<Typography variant="h5" align="center" gutterBottom>
+										The goal of the Code Academy is to accelerate the growth of
+										globally competitive coding talent in Palestine. We believe
+										that by helping Palestinian youth build high-tech and modern
+										skills to participate in the exponentially growing tech
+										sector, they will have a sustainable and long-term pathway
+										to high-quality employment.
+									</Typography>
```
Comment (leggomuhgreggo) [inline client/src/Components/About/index.js]: I'm not sure heading is the most appropriate HTML semantics. A paragraph or span would probably make more sense, but not a huge concern

File: client/src/Components/About/style.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	link: {
+		textDecoration: 'unset',
+		color: 'white',
+		textTransform: 'capitalize',
+		'&:active': {
+			color: 'white',
+		},
+	},
+}));
+export default styles;
```
Comment (leggomuhgreggo) [inline client/src/Components/About/style.js]: A cool improvement might be to define breakpoints and theme colors in the theme or some other config.

File: client/src/Components/Cards/index.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
+import React, { Component, Fragment } from 'react';
+import { Paper, Grid, Avatar, Typography, Divider } from '@material-ui/core';
+import { withStyles } from '@material-ui/styles';
+import styles from './style';
+class Cards extends Component {
+	state = {};
```
Comment (leggomuhgreggo) [inline client/src/Components/Cards/index.js]: Is this used?

File: client/src/Components/Cards/index.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+	render() {
+		const { classes, location } = this.props;
+		return (
+			<Fragment>
+				<Grid
+					classes={{ root: classes.root }}
+					className={location === '/user/:userId' ? classes.card : null}
```
Comment (leggomuhgreggo) [inline client/src/Components/Cards/index.js]: I could see abstracting this logic to a method being useful long term.

File: client/src/Components/Cards/index.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+				>
+					<Paper
+						variant="outlined"
+						elevation={1}
+						classes={{ root: classes.paper }}
+					>
+						<Avatar
+							alt="github"
```
Comment (leggomuhgreggo) [inline client/src/Components/Cards/index.js]: Nice! Glad to see the alt text!

File: client/src/Components/Cards/style.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
+const Styles = (theme) => ({
+	root: {
+		display: 'flex',
+		flexWrap: 'wrap',
+		'& > *': {
+			margin: theme.spacing(1),
+			width: theme.spacing(16),
+			height: theme.spacing(16),
+		},
```
Comment (leggomuhgreggo) [inline client/src/Components/Cards/style.js]: Nice use of theme methods!

File: client/src/Components/Collapsible/index.js [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+export default function Collapsible(props) {
+	const [expanded, setExpanded] = React.useState('panel1');
+
+	const handleChange = (panel) => (event, newExpanded) => {
+		setExpanded(newExpanded ? panel : false);
+	};
+	const classes = useStyles();
+
+	const { location } = props;
+	console.log(location);
```
Comment (leggomuhgreggo) [inline client/src/Components/Collapsible/index.js]: 👋

File: client/src/Components/Collapsible/index.js [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+import Typography from '@material-ui/core/Typography';
+import AddBoxIcon from '@material-ui/icons/AddBox';
+import IndeterminateCheckBoxIcon from '@material-ui/icons/IndeterminateCheckBox';
+import useStyles from './style';
+import { Button } from '@material-ui/core';
+import Linkify from 'react-linkify';
+import Cards from '../Cards';
+
+export default function Collapsible(props) {
+	const [expanded, setExpanded] = React.useState('panel1');
```
Comment (leggomuhgreggo) [inline client/src/Components/Collapsible/index.js]: This seems like it's being used as an identifier? Might consider making that more clear. Took me a second to understand the intention.

File: client/src/Components/Collapsible/index.js [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+	const { location } = props;
+	console.log(location);
+
+	return (
+		<Fragment>
+			<MuiExpansionPanel
+				square
+				expanded={expanded === `panel${props.id}`}
+				onChange={handleChange(`panel${props.id}`)}
```
Comment (leggomuhgreggo) [inline client/src/Components/Collapsible/index.js]: This totally works! Might be fun to think about other ways to handle this logic in an abstraction to reduce the dependence on the "magic string"

File: client/src/Components/Collapsible/index.js [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+				classes={{
+					root: classes.ExpansionPanelroot,
+					expanded: classes.ExpansionPanelexpanded,
+				}}
+			>
+				<MuiExpansionPanelSummary
+					aria-controls={`panel${props.id}bh-content`}
```
Comment (leggomuhgreggo) [inline client/src/Components/Collapsible/index.js]: Nice use of aria attributes!

File: client/src/Components/Collapsible/index.js [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+								<Cards location={location} />
+								{props.body}
+								<Button
+									variant="contained"
+									color="secondary"
+									className={classes.btn}
+									onClick={props.onClick(props.id)}
+									style={{ display: props.id === 2 ? 'none' : 'unset' }}
+								>
+									{props.id === 5 ? 'Submit' : 'Check Out'}
```
Comment (leggomuhgreggo) [inline client/src/Components/Collapsible/index.js:67]: What is id 5?  Might be more clear if you used a conditional helper method like `isCheckoutPanel(panelId)`

File: client/src/Components/Collapsible/style.js [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+	},
+	bodyText: {
+		top: 127,
+		left: 105,
+		color: 'rgba(0, 0, 0, 0.79)',
+		width: 896,
+		height: 96,
+		position: 'absolute',
+		fontSize: 30,
+		// color: 'rgba(0, 0, 0, 0.79)',
```
Comment (leggomuhgreggo) [inline client/src/Components/Collapsible/style.js]: 👋

File: client/src/Components/Collapsible/style.js [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+	bigContent: {
+		height: 789,
+	},
+	ExpansionPanelroot: {
+		width: 1017,
+		marginBottom: 60,
+		border: '1px solid #4CB7A7',
+		boxShadow: 'none',
+		'&:not(:last-child)': {
+			// borderBottom: 1,
```
Comment (leggomuhgreggo) [inline client/src/Components/Collapsible/style.js]: 👋

File: client/src/Components/Dialog/index.js [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+class AlertDialog extends Component {
+	state = { open: true };
+
+	handleClickOpen = () => {
+		this.setState({ open: true });
+	};
+
+	handleClose = () => {
+		this.setState({ open: false });
+	};
```
Comment (leggomuhgreggo) [inline client/src/Components/Dialog/index.js:18]: Very clean 👍

File: client/src/Components/DropMenu/index.js [hunk 18]
```diff
@@ -0,0 +1,117 @@
...
+		this.setState({ anchorEl: null });
+	};
+
+	handleMenuClick = (event, index) => {
+		if (index === 1) {
+			return this.props.history.push('/user/:userId/profile');
+		} else if (index === 3) {
+			return this.props.history.push('/');
+		} else this.setState({ open: true });
+		event.preventDefault();
```
Comment (leggomuhgreggo) [inline client/src/Components/DropMenu/index.js:61]: Usually preventDefault would be first, but maybe there's a reason it not here.

Fun fact: If you're returning from the precedent conditional block you can skip the `else` and simplify the cascading logic.

File: client/src/Components/DropMenu/index.js [hunk 19]
```diff
@@ -0,0 +1,117 @@
...
+		anchorEl: null,
+		open: false,
+	};
+
+	handleClick = (event) => {
+		this.setState({ anchorEl: event.currentTarget });
+	};
+
+	handleClose = () => {
+		console.log('here');
```
Comment (leggomuhgreggo) [inline client/src/Components/DropMenu/index.js:51]: 👋

File: client/src/Components/LoginForm/index.js [hunk 20]
```diff
@@ -0,0 +1,80 @@
+import React, { Fragment, Component } from 'react';
+import { TextField } from '@material-ui/core';
+
+class TextFieldValidation extends Component {
+	state = {};
+	render() {
+		let { onChange, error } = this.props;
+		return (
+			<Fragment>
```
Comment (leggomuhgreggo) [inline client/src/Components/LoginForm/index.js]: Good use of Fragment 👍 

Relatively recently facebook added support for Fragments to be written like `<>...</>` which skips the need to import, but nothing wring with using the more explicit component.

File: client/src/Components/LoginForm/index.js [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+					<h6>email or password incorrect</h6>
+				) : (
+					<h6></h6>
+				)}
+			</Fragment>
+		);
+	}
+}
+
+export default TextFieldValidation;
```
Comment (leggomuhgreggo) [inline client/src/Components/LoginForm/index.js]: Very readable! I could see potentially abstracting the TextField if this file were going to change a lot, but this is very easy to understand.

File: client/src/Components/MenuBar/index.js [hunk 22]
```diff
@@ -0,0 +1,23 @@
...
+import MenuIcon from '@material-ui/icons/Menu';
+import { IconButton } from '@material-ui/core';
+
+const MenuBar = (props) => {
+	const classes = myStyles();
+	return (
+		<Fragment>
+			<IconButton
+				edge="end"
+				aria-label="menu"
```
Comment (leggomuhgreggo) [inline client/src/Components/MenuBar/index.js]: 🚀

File: client/src/Components/MenuBar/style.js [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+
+const useStyles = makeStyles((theme) => ({
+	btnMenu: {
+		[theme.breakpoints.between('0', '1094')]: {
+			display: 'display !important',
+		},
+
+		[theme.breakpoints.between('1095', 'xl')]: {
+			display: 'none !important',
+		},
```
Comment (leggomuhgreggo) [inline client/src/Components/MenuBar/style.js]: This style must be very important 😄

File: client/src/Components/ShareDialog/style.js [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+		[theme.breakpoints.down('425')]: {
+			width: 200,
+			height: 60,
+			fontSize: 20,
+		},
+		[theme.breakpoints.down('382')]: {
+			width: 150,
+			height: 50,
+			fontSize: 15,
+		},
```
Comment (leggomuhgreggo) [inline client/src/Components/ShareDialog/style.js]: Just wanted to share that some shops adhere to a "mobile first" convention with breakpoint styles, where the base styles are for mobile and then larger breakpoints override with `min-width` or `breakpoint.up`. But this is totally legitimate too!

File: client/src/Components/Stories/index.js [hunk 25]
```diff
@@ -0,0 +1,10 @@
+import React, { Component } from 'react';
+
+class Stories extends Component {
+	state = {};
+	render() {
+		return <h1>this is a stories</h1>;
+	}
+}
```
Comment (leggomuhgreggo) [inline client/src/Components/Stories/index.js]: This seems like a great opportunity for Stateless Functional Component (SFC) 😄

File: client/src/Components/TemporaryDrawer/index.js [hunk 26]
```diff
@@ -0,0 +1,178 @@
...
+
+import MenuBar from '../MenuBar';
+
+export default function TemporaryDrawer(props) {
+	const classes = useStyles();
+	const [state, setState] = React.useState({
+		right: false,
+		link: '/',
+	});
+	console.log(props);
```
Comment (leggomuhgreggo) [inline client/src/Components/TemporaryDrawer/index.js]: 👋

File: client/src/Components/Textfields/index.js [hunk 27]
```diff
@@ -0,0 +1,90 @@
...
+					>
+						{this.state.Submitted ? 'Submitted' : 'Upload'}
+					</Button>
+				</div>
+			</Fragment>
+		);
+	}
+}
+
+export default withStyles(style)(Textfields);
```
Comment (leggomuhgreggo) [inline client/src/Components/Textfields/index.js]: 👏 👏 👏

File: client/src/Content/firstPage.js [hunk 28]
```diff
@@ -0,0 +1,54 @@
...
+				'1- First click this link to join the class' +
+				'\n' +
+				'https://gsg-code-academy.typingclub.com/' +
+				'\n' +
+				'2- Click Enter a Class Code \n' +
+				'3- Copy and paste this code (GMUR84A) then click Sign up after that you can fill in your information\n' +
+				'\n' +
+				'By the application deadline, you must complete at least the first 50 lessons in Typing Club.\n' +
+				'Each lesson should take less than 10 minutes to\n' +
+				'complete. The best way to improve your typing is to do a small amount, practising every day, rather than doing all of the lessons at once.',
```
Comment (leggomuhgreggo) [inline client/src/Content/firstPage.js]: I wonder if template literals would work? They might alleviate the need for the `\n` action.

File: client/src/Screens/Accounts/index.js [hunk 29]
```diff
@@ -0,0 +1,84 @@
...
+		return this.setState({ data: Content() });
+	}
+
+	scrollToTop = () => {
+		scroll.scrollToTop();
+	};
+
+	handleClickOpen = (id) => (event) => {
+		// this.props.history.push('/');
+		// console.log(this.props, id);
```
Comment (leggomuhgreggo) [inline client/src/Screens/Accounts/index.js]: 👋

**Review by leggomuhgreggo — APPROVED** [review]
Heroic work by the team! Way to bring it together Karmel!

I wasn't able to get through all of the files this time around. Ideally these changes would be broken up into multiple PRs but I know that isn't always easy to do when you're in the zone on large scale integration work. 

Have these files had prettier / eslint run against them? The indentation was testing the limited of GitHub's split view. 

Most of my comments are kind "Food for thought" considerations, or highlighting dev artifacts like commented code and log statements. It really was very easy to read and understand for the most part but I think there are areas where the logic might improve from abstraction.

I know there's more y'all are excited to do, but I wanted to say that the app really is very nice. I can't tell you how much better it is than the average quality of work from code bootcamps in the US. 

Looking forward to seeing the updates and checking out the other PRs later today 👍

File: .gitignore (hunk 1 above)
Comment (karmelyoei) [reply .gitignore]: yeah there is a small conflict here

File: client/.gitignore (hunk 2 above)
Comment (karmelyoei) [reply client/.gitignore]: Yeah I put it in either ways but still see this file so I have to delete it

File: client/src/Components/About/index.js (hunk 4 above)
Comment (karmelyoei) [reply client/src/Components/About/index.js]: right!

File: client/src/Components/Cards/index.js (hunk 6 above)
Comment (karmelyoei) [reply client/src/Components/Cards/index.js]: what do you mean the cards or the state? if the cards yeah i used them on the profile section if the state na just

File: client/src/Components/Cards/index.js (hunk 7 above)
Comment (karmelyoei) [reply client/src/Components/Cards/index.js]: How ?

File: client/src/Components/Collapsible/index.js (hunk 10 above)
Comment (karmelyoei) [reply client/src/Components/Collapsible/index.js]: I should get ride of all consoles

File: client/src/Components/Collapsible/index.js (hunk 11 above)
Comment (karmelyoei) [reply client/src/Components/Collapsible/index.js]: Oh man this is a Hook style for a function component kind hard to understand  I am not into it either

File: client/src/Components/Collapsible/index.js (hunk 14 above)
Comment (karmelyoei) [reply client/src/Components/Collapsible/index.js:67]: Yeah each Content has an id for this collapsible  should have a submit button, not a checkout, so i send the id of the contents as a prop so I can manipulate with the dom element display

File: client/src/Components/ShareDialog/style.js (hunk 24 above)
Comment (karmelyoei) [reply client/src/Components/ShareDialog/style.js]: yeah agree but we had to start with the website first cause we don't have time

File: client/src/Components/Stories/index.js (hunk 25 above)
Comment (karmelyoei) [reply client/src/Components/Stories/index.js]: this section related to my colleague don't worry about it

File: client/src/Content/firstPage.js (hunk 28 above)
Comment (karmelyoei) [reply client/src/Content/firstPage.js]: whats that ?

**Pull request closed without merging**